light-client/relayer/tests/block_processing.rs

```diff
@@ -0,0 +1,219 @@
+                block.id.clone(),
+                Default::default(),
+                Default::default(),
+                ResponderId::from_str(&format!("node{idx}.test.com:8433")).unwrap(),
+            );
+            BlockMetadata::from_contents_and_keypair(bmc, signer).unwrap()
+        })
+        .collect();
+
+    ledger
+        .append_block(
+            &block,
+            &block_contents,
+            None,
+            Some(&block_metadata[0].clone()),
+        )
+        .unwrap();
+    (block, block_contents, block_metadata)
+}
+
+#[test_with_logger]
+fn test_relayer_processing(logger: Logger) {
+    let mut rng: StdRng = SeedableRng::from_seed([216u8; 32]);
+
+    let blockchain_path = TempDir::new()
+        .expect("Could not make tempdir for blockchain state")
+        .into_path();
+
+    let watcher_db_path = {
+        // Set up the Watcher db.
+        let watcher_db_path = blockchain_path.join("watcher");
+        std::fs::create_dir(&watcher_db_path).expect("failed to create watcher dir");
+        WatcherDB::create(&watcher_db_path).expect("failed to create WarcherDB");
+        watcher_db_path
+    };
+
+    // Set up an empty ledger db.
+    let ledger_db_path = {
+        let ledger_db_path = blockchain_path.join("ledger_db");
+        std::fs::create_dir(&ledger_db_path).expect("failed to create ledger dir");
+        LedgerDB::create(&ledger_db_path).expect("failed to create LedgerDB");
+        ledger_db_path
+    };
+
+    let mut ledger = LedgerDB::open(&ledger_db_path).unwrap();
+    let tx_source_urls: Vec<Url> = (0..5)
+        .filter_map(|idx| Url::parse(&format!("https://node{idx}.test.com:8433")).ok())
+        .collect();
+
+    let watcher = WatcherDB::open_rw(&watcher_db_path, &tx_source_urls, logger.clone())
+        .expect("Could not create watcher_db");
+
+    // Initialize ledger
+    let sender = AccountKey::random(&mut rng);
+    let num_blocks = 1;
+    initialize_ledger(BLOCK_VERSION, &mut ledger, num_blocks, &sender, &mut rng);
+
+    // Make nodes and signer identities
+    let mut signers: Vec<Ed25519Pair> = Vec::new();
+    let mut quorum_nodes: Vec<QuorumSetMember<NodeID>> = Vec::new();
+    for idx in 0..5 {
+        let (node_id, signer) = test_node_id_and_signer(idx);
+        signers.push(signer);
+        quorum_nodes.push(QuorumSetMember::Node(node_id));
+    }
+    
+    let config = Config {
+        start_block_index: 1,
+        min_signatures: 4,
+        ledger_db: ledger_db_path,
+        watcher_db: watcher_db_path,
+        admin_listen_uri: None,
+    };
+
+    log::info!(logger, "Starting relayer");
+    let sender = TestSender {
+        logger: logger.clone(),
+        sent: Default::default(),
+    };
+
+    let known_valid_block_ids = BTreeSet::default();
+    let current_tvs = TrustedValidatorSet {
+        quorum_set: QuorumSet::new(config.min_signatures, quorum_nodes),
+    };
+
+    let verifier = LightClientVerifier {
```

Comment:
If you're using the real verifier here then is there a reason for the `TestVerifier` and the `Verifier` trait at all?